ClientCore/I18N/Locale.cs

```diff
@@ -0,0 +1,269 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Reflection.Metadata;
+using Rampastring.Tools;
+using Rampastring.XNAUI.XNAControls;
+
+namespace ClientCore.I18N;
+
+public class Locale : ICloneable
```

Comment:
Locale means a specific language, region, formatting settings.
It does not stand for a translation file for a locale.
e.g. en-US is a locale, while a text written in en-US is not a locale

Reply:
It's not only for text, I store size/position INI values there too (and assets too). Locale is short and concise name for that.

Reply:
I mean, in the whole world, there is only one `locale` identified as en-US.
This word is only used for specifiying language, region, formatting settings. Not for any instances that written in a language.
Don't misuse it, especially for INI key names and file names.
Locale assets, or Localization should be fine.

Reply:
> Locale assets

Redundancy...

>  Localization

That is a process.

I fail to see a problem in calling locale-specific stuff just locale, it identifies what the class is supposed to contain and do just fine.

Reply:
> > Locale assets
> 
> Redundancy...
> 
> > Localization
> 
> That is a process.
> 
> I fail to see a problem in calling locale-specific stuff class just `Locale`, it identifies what the class is supposed to contain and do just fine.

Example: https://developer.playcanvas.com/api/pc.I18n.html
where 'localization data' and 'asset' are used.
`this.app.i18n.addData()` loads a specific translation and other stuffs

Reply:
On the second thought what you suggested as LocaleAssets is already kinda is what I have, the `Locale` has `Values` (name pending, that is pretty much your assets, data or w/e). That is pretty much equivalent to `Assets`. So, `LocaleAssets.Assets`?..

Reply:
The `Locale` class Contains `Values` or `Assets`, which means the class is not a locale but a collection of locale assets. It is strange that a locale class has a property named locale or locale code which refers to the real locale.
`LocaleAssets.Values` or `LocaleAssets.Items` for translation items and `LocaleAssets.Locale` for accessing the locale that it refers to. It is pretty fine.

Reply:
> a locale class has a property named locale or locale code which refers to the real locale.

Fair point, I'll change that to just `Code`.

> The Locale class Contains Values or Assets, which means the class is not a locale but a collection of locale assets.

That I disagree with. The class is an entity, and each instance of that entity contains some data specific to it.

Reply:
> That I disagree with. The class is an entity, and each instance of that entity contains some data specific/related to it.

Disagreed. Imagine a class named `Language` but it does not refers to `English` but all kinds of stuffs written in English. This is messy.

The class name should matches its content instead of an entity. For example, each `Map` has its `GameMode`, so we have two classes `Map` and `GameMode`. You can expect 'Map.GameMode' but not `GameMode.Maps`. When designing `GameMode`, there is no such thing as a map, while designing `Map`, one can assume `GameMode` is availble as `GameMode` is naturally something that should belongs to a `Map`. Then `Map` classes depends on `GameMode` but NOT vice versa, so one can store `GameMode` inside a `Map`. In some programming languages like Go, circular dependencies are prohibited. 

Now think the relationship between `Locale` and `Asset`. A `Locale` is defined directly depending on language, region, formats, etc., and it is like a tag (something like a `GameMode`). `Asset` on the otherhand, natually has the language and format, i.e. each `Asset` has its `Locale`.  Therefore, `Asset` (or `Assets`) classes depends on `Locale` but NOT vice versa. In this case, even if we can have an one-to-many relationship between `Locale` and `Asset`, it is still not a good way to store `Assets` in `Locale`. Such circular dependencies brings mess and ambiguity.

Reply:
Besides, think how Qt handles the relationships of each concept:

a `QTranslator` object can load a `Translation`
a `QTranslator` object can automatically load a Translation according to the user's preferred UI display `Locale`

The relationship among `QTranslator`, `Translation`, and `Locale` is very clear. `Locale` is just a tag.

Even if the word `Translation` is mainly for textes instead of UI locations, it does have a clear relationship and the concept of `Locale` is not messed up. Therefore we should use `TranslationAssets` or `LocaleAssets` for the class name.